cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/steps/AsyncLoopSteps.java

```diff
@@ -95,20 +90,15 @@ public void addBranches(@Param(ScoreLangConstants.ASYNC_LOOP_STATEMENT_KEY) Asyn
                         nextStepId,
                         branchBeginStep);
 
-                fireEvent(
-                        executionRuntimeServices,
-                        ScoreLangConstants.EVENT_BRANCH_START,
-                        "async loop branch created",
-                        runEnv.getExecutionPath().getCurrentPathPeekForward(),
-                        Pair.of(ScoreLangConstants.REF_ID, refId),
-                        Pair.of(RuntimeConstants.SPLIT_ITEM_KEY, splitItem),
-                        Pair.of(LanguageEventData.levelName.TASK_NAME.name(), nodeName));
+                fireEvent(executionRuntimeServices, ScoreLangConstants.EVENT_BRANCH_START,
+                        "async loop branch created", runEnv.getExecutionPath().getCurrentPath(),
+                        LanguageEventData.StepType.TASK, nodeName, Pair.of(ScoreLangConstants.REF_ID, refId),
+                        Pair.of(RuntimeConstants.SPLIT_ITEM_KEY, splitItem));
+
+                runEnv.getExecutionPath().forward();
```

Comment:
In the last branch, is it a problem that the path changes? Is there any chance that it will be used somewhere?